// @dhis2/app-adapter/build/es/components/AppWrapper.js
import _JSXStyle from "styled-jsx/style";
import { HeaderBar } from '@dhis2/ui';
import PropTypes from 'prop-types';
import React from 'react';
import { useCurrentUserLocale } from '../utils/useLocale.js';
import { useVerifyLatestUser } from '../utils/useVerifyLatestUser.js';
import { Alerts } from './Alerts.js';
import { ErrorBoundary } from './ErrorBoundary.js';
import { LoadingMask } from './LoadingMask.js';
import PWAUpdateManager from './PWAUpdateManager.js';
import { styles } from './styles/AppWrapper.style.js';
export const AppWrapper = ({
  appName,
  children,
  offlineInterface
}) => {
  const {
    loading: localeLoading
  } = useCurrentUserLocale();
  const {
    loading: latestUserLoading
  } = useVerifyLatestUser();

  if (localeLoading || latestUserLoading) {
    return /*#__PURE__*/React.createElement(LoadingMask, null);
  }

  return /*#__PURE__*/React.createElement("div", {
    className: `jsx-${styles.__hash}` + " " + "app-shell-adapter"
  }, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: styles.__hash
  }, styles), /*#__PURE__*/React.createElement(HeaderBar, {
    appName: appName
  }), /*#__PURE__*/React.createElement("div", {
    className: `jsx-${styles.__hash}` + " " + "app-shell-app"
  }, /*#__PURE__*/React.createElement(ErrorBoundary, {
    onRetry: () => window.location.reload()
  }, children)), /*#__PURE__*/React.createElement(Alerts, null), /*#__PURE__*/React.createElement(PWAUpdateManager, {
    offlineInterface: offlineInterface
  }));
};
AppWrapper.propTypes = {
  appName: PropTypes.string.isRequired,
  offlineInterface: PropTypes.object.isRequired,
  children: PropTypes.node
};
